_posts/2019-05-31-ucx-dgx.md

```diff
@@ -353,6 +353,39 @@ def DGX(
     )
 ```
 
+Leveraging the `DGX` configuration can produce significant performance gains for communication-intensive applications.  To illustrate this point, we have modified the [SVD Dask-GPU](https://github.com/mrocklin/dask-gpu-benchmarks/blob/master/cupy-svd.ipynb) benchmark to use the UCX protocol for inter-process communication. Here is a minimal implementation of the UCX-enabled code:
+
+```python
+import cupy
+import dask
+import dask.array
+from dask.distributed import Client, wait
+from dask_cuda import DGX
+
+# Define DGX cluster and client
+cluster = DGX(CUDA_VISIBLE_DEVICES=[4,5,6,7])
+client = Client(cluster)
+
+# Create random data
+rs = dask.array.random.RandomState(RandomState=cupy.random.RandomState)
+x = rs.random((1000000, 1000), chunks=(10000, 1000))
+x = x.persist()
+
+# Perform distributed SVD
+u, s, v = dask.array.linalg.svd(x)
+u, s, v = dask.persist(u, s, v)
+_ = wait([u, s, v])
+```
+
+By defining a DGX cluster, and therefore enabling UCX, the overall communication time is reduced by almost an order of magnitude. To produce the task-stream figures below, the benchmark was run on a DGX-1 with `CUDA_VISIBLE_DEVICES=[0,1,2,3]`.  It is clear that the red task bars, corresponding to inter-process communication, are significantly compressed. 
+
+**Before UCX**:
+<iframe src="https://raw.githubusercontent.com/rjzamora/raw-host/master/task_stream_lcc_dgx16.html" width="800" height="400"></iframe>
```

Comment:
FYI for this to work the HTML needs to be served somewhere.  I had to put these plots files into a `gh-pages` branch so that github would host them, so that the user's browser would run the javascript contained within.

Reply:
Right, I forgot about this - The files are also hosted at `https://rjzamora.github.io/raw-host/<file-name>` now, but thanks for taking care of it!